UglyTicTacToe/Default.aspx.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.UI;
+using System.Web.UI.WebControls;
+using UglyTicTacToe.Models;
+using System.Windows;
+namespace UglyTicTacToe
+{
+    public partial class Default : System.Web.UI.Page
+    {
+        MyState lastState = Buttons.recentState;
+        MyState currentState;
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            if (!IsPostBack)
+            {
+                GameLogic.initArray();
+            }
+        }
+
+        protected void NewGameButton_Click(object sender, EventArgs e)
+        {
+            Buttons.recentState = MyState.None;
+            GameLogic.ButtonsLeft = 9;
+            GameLogic.initArray();
+            Response.Redirect("~/Default.aspx");
+        }
+
+        protected void Button_Click(object sender, EventArgs e)
+        {
+            Button clickedButton = (Button)sender;
+            string buttonIDString = "";
+            clickedButton.Enabled = false;
+            GameLogic.ButtonsLeft -= 1;
+            buttonIDString = clickedButton.ID;
+            if (lastState == MyState.None || lastState == MyState.Nill)
+            {
+                clickedButton.Text = "X";
+                currentState = MyState.Cross;
+                Buttons.recentState = currentState;
+            }
+            else if (lastState == MyState.Cross)
+            {
+                clickedButton.Text = "O";
+                currentState = MyState.Nill;
+                Buttons.recentState = currentState;
+            }
+
+            UpdateGameLogic(buttonIDString);
+
+            if (GameLogic.ApplyLogic() == MyState.Cross)
+            {
+                WinNotificationX.Visible = true;
+                Confetti.Visible = true;
+                TTT.Visible = true;
+                DisableAllButtons();
+            }
+            else if (GameLogic.ApplyLogic() == MyState.Nill)
+            {
+                WinNotificationO.Visible = true;
+                Confetti.Visible = true;
+                TTT.Visible = true;
+                DisableAllButtons();
+            }
+            else if (GameLogic.ButtonsLeft == 0)
+            {
+                //There is a tie
+                TieNotification.Visible = true;
+            }
+            // if win (display who wins and disable all buttons)
+        }
+
+        public void UpdateGameLogic(string buttonID)
```

Comment:
Instead of passing the button ID to this method, you could pass the button's `CommandArgument` property value, which is already being set to the appropriate array indexes. Each command argument value is in the format of `{index},{index}`, so you'd parse out the first and second indexes, convert them to the `int` type, and use those values to index into your `GameLogic.buttonArray` 2D array. That'd allow you to remove the below `switch` statement.